hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/FileIndex.java

```diff
@@ -145,7 +149,7 @@ public FileStatus[] getFilesInPartitions() {
       return new FileStatus[0];
     }
     String[] partitions = getOrBuildPartitionPaths().stream().map(p -> fullPartitionPath(path, p)).toArray(String[]::new);
-    FileStatus[] allFiles = FSUtils.getFilesInPartitions(HoodieFlinkEngineContext.DEFAULT, metadataConfig, path.toString(), partitions)
+    FileStatus[] allFiles = FSUtils.getFilesInPartitions(hoodieFlinkEngineContext, metadataConfig, path.toString(), partitions)
```

Comment:
What kind of hadoop configuration do you wanna to pass around?

Reply:
Something like that
<img width="761" alt="image" src="https://user-images.githubusercontent.com/46181516/236088448-e9fdd856-ffc2-4edd-b790-3ea80f4caa96.png">